types/index.d.ts

```diff
@@ -222,16 +244,20 @@ export interface SmartActionLoadHookField extends SmartActionHookField {
   position: number,
 }
 
+export interface SmartActionHookRequest extends ForestRequest {
+  body: SelectAllRequestBody,
+}
+
 export interface SmartActionLoadHook<T = any> {
-  (context: { fields: SmartActionLoadHookField[], record: T & Document }): SmartActionLoadHookField[]
+  (context: { fields: SmartActionLoadHookField[], request: SmartActionHookRequest }): SmartActionLoadHookField[]
```

Comment:
Actually, the T generic type is not used anymore. If we don't provide any record to the user, you can remove it.